src/Common/src/System/Net/CookieParser.cs

```diff
@@ -0,0 +1,892 @@
+                            switch (_tokenizer.Token)
+                            {
+                                case CookieToken.Domain:
+                                    if (!domainSet)
+                                    {
+                                        domainSet = true;
+                                        cookie.Domain = CheckQuoted(_tokenizer.Value);
+                                        IsQuotedDomainField.SetValue(cookie, _tokenizer.Quoted);
+                                    }
+                                    break;
+
+                                case CookieToken.Path:
+                                    if (!pathSet)
+                                    {
+                                        pathSet = true;
+                                        cookie.Path = _tokenizer.Value;
+                                    }
+                                    break;
+
+                                case CookieToken.Port:
+                                    if (!portSet)
+                                    {
+                                        portSet = true;
+                                        try
+                                        {
+                                            cookie.Port = _tokenizer.Value;
+                                        }
+                                        catch (CookieException)
+                                        {
+                                            // This cookie will be rejected
+                                            InternalSetNameMethod.Invoke(cookie, EmptyStringArray);
+                                        }
+                                    }
+                                    break;
+
+                                case CookieToken.Version:
+                                    // this is a new cookie, this token is for the next cookie.
+                                    _savedCookie = new Cookie();
+                                    int parsed;
+                                    if (int.TryParse(_tokenizer.Value, out parsed))
+                                    {
+                                        _savedCookie.Version = parsed;
+                                    }
+                                    return cookie;
+
+                                case CookieToken.Unknown:
+                                    // this is a new cookie, the token is for the next cookie.
+                                    _savedCookie = new Cookie();
+                                    InternalSetNameMethod.Invoke(_savedCookie, new object[] { _tokenizer.Name });
+                                    _savedCookie.Value = _tokenizer.Value;
+                                    return cookie;
+                            }
+                            break;
+
+                        case CookieToken.Attribute:
+                            switch (_tokenizer.Token)
```

Comment:
Instead of a switch for one only one case better `if (_tokenizer.Token == CookieToken.Port)`